File: flutter_app/lib/models/device_session.dart
class DeviceSession {
  final String id;
  final String deviceId;
  final String deviceName;
  final String platform;
  final String appVersion;
  final DateTime lastActiveAt;

  const DeviceSession({
    required this.id,
    required this.deviceId,
    required this.deviceName,
    required this.platform,
    required this.appVersion,
    required this.lastActiveAt,
  });

  factory DeviceSession.fromJson(Map<String, dynamic> json) {
    return DeviceSession(
      id: (json['_id'] ?? '').toString(),
      deviceId: (json['deviceId'] ?? '').toString(),
      deviceName: (json['deviceName'] ?? 'Unknown device').toString(),
      platform: (json['platform'] ?? 'unknown').toString(),
      appVersion: (json['appVersion'] ?? '1.0.0').toString(),
      lastActiveAt: json['lastActiveAt'] != null
          ? DateTime.parse(json['lastActiveAt']).toLocal()
          : DateTime.now(),
    );
  }
}

File: backend/models/DeviceSession.js
const mongoose = require('mongoose');

const deviceSessionSchema = new mongoose.Schema(
  {
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    deviceId: { type: String, required: true },
    deviceName: { type: String, default: 'Unknown device' },
    platform: { type: String, default: 'unknown' },
    appVersion: { type: String, default: '1.0.0' },
    isActive: { type: Boolean, default: true },
    lastActiveAt: { type: Date, default: Date.now },
  },
  { timestamps: true }
);

deviceSessionSchema.index({ user: 1, deviceId: 1 }, { unique: true });

module.exports = mongoose.model('DeviceSession', deviceSessionSchema);

